radixdlt/src/main/java/com/radixdlt/consensus/ConsensusRunner.java

```diff
@@ -65,15 +71,20 @@ public String toString() {
 
 	private final ConnectableObservable<Event> events;
 	private final Object lock = new Object();
+	private final VertexStore vertexStore;
 	private Disposable disposable;
 
 	@Inject
 	public ConsensusRunner(
 		EpochRx epochRx,
 		EventCoordinatorNetworkRx network,
 		PacemakerRx pacemakerRx,
-		EpochManager epochManager
+		LocalSyncRx localSyncRx,
+		SyncVerticesRPCRx rpcRx,
+		EpochManager epochManager,
+		VertexStore vertexStore //TODO: remove this since it should only be provided by Epoch manager

```

Comment:
Why should the vertex store be provided by the epoch manager?

Reply:
A BFT run with fixed validatorSet from start to end is an "epoch". Thus VertexStore which is specifically used by BFT should have a lifecycle maintained by the epoch.